WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanFragment.kt

```diff
@@ -3,26 +3,68 @@ package org.wordpress.android.ui.jetpack.scan
 import android.os.Bundle
 import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
+import kotlinx.android.synthetic.main.scan_fragment.recycler_view
 import org.wordpress.android.R
 import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.ui.jetpack.scan.ScanViewModel.UiState.Content
+import org.wordpress.android.ui.jetpack.scan.adapters.ScanAdapter
+import org.wordpress.android.ui.utils.UiHelpers
 import javax.inject.Inject
 
 class ScanFragment : Fragment(R.layout.scan_fragment) {
+    @Inject lateinit var uiHelpers: UiHelpers
     @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
     private lateinit var viewModel: ScanViewModel
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        (requireActivity().application as WordPress).component().inject(this)
-    }
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        initViewModel()
+
```

Comment:
Idea (💡 ): This is just me proposing one of those readability things that I like advocating in favor (😊 ), but this is very opinionated, so please consider it as such. What if the `onViewCreated(...)` function looked like that, do you find the below more readable?

```
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initDagger()
        initAdapter()
        initViewModel(getSite(savedInstanceState))
    }
```

`initDagger()` can be extracted in its own function using the `requireActivity()` directly. Then, the site state related code can be extracted to another private function as well and used directly through `getSite(...)`. Here again, `requireActivity()` can be used directly to avoid creating an extra variable. Wdyt?

Reply:
In 01129a8